src/components/Headers/DefaultHeader/DefaultHeader.tsx

```diff
@@ -1,25 +1,30 @@
-'use client'
 import React from 'react'
-import { Button } from '@/components/Button/Button'
 import styles from './DefaultHeader.module.scss'
 import logo from '@/assets/images/logo.png'
-import GithubIcon from '@/assets/icons/GithubIcon'
 import Link from 'next/link'
-import { useSession, signIn } from 'next-auth/react'
-import { useRouter } from 'next/navigation'
 import Image from 'next/dist/client/image'
-import { AppRoutes } from '@/utils/routes'
-import AddIcon from '@/assets/icons/AddIcon'
+import SignInWithGithubBtn from '@/components/SignInWithGithubBtn/SignInWithGithubBtn'
+import CreateProfileBtn from '@/components/CreateProfileBtn/CreateProfileBtn'
+import MyProfileBtn from '@/components/MyProfileBtn/MyProfileBtn'
 
-const DefaultHeader = () => {
-  const { status, data: session } = useSession()
+type SessionType = {
+  user: {
+    name: string
+    image: string
+    profileId: string | null
+  }
+}
+
+interface DefaultHeaderProps {
+  session: SessionType | null
+}
+
+const DefaultHeader = ({ session }: DefaultHeaderProps) => {
```

Comment:
**nitpick:** Why passing down this session as prop and not `getServerSession` here in this component directly?